transformer_engine/pytorch/attention.py

```diff
@@ -3376,7 +3380,10 @@ def __init__(
             set_all_rng_states(self.rng_states_tracker.get_states())
             attention_dropout_ctx = self.rng_states_tracker.fork
 
-        norm_factor = math.sqrt(kv_channels)
+        if softmax_scale is None:
+            norm_factor = math.sqrt(kv_channels)
+        else:
+            norm_factor = softmax_scale
```

Comment:
The quickest way to keep the `softmax_scale` kwarg consistent with Flash Attention:
```suggestion
        if softmax_scale is None:
            softmax_scale = math.sqrt(kv_channels)
        norm_factor = 1 / softmax_scale
```

Reply:
I am ok with this implementation, not so sure whether this is a wanted implementation for TE's code readability though. What do you think

Reply:
My thinking is that it's more important to get the API right. If we add something non-standard like `norm_factor`, then it'll confuse users and be painful to remove in the future. In the future, I think it would be nice to refactor to use `softmax_scale` instead of `norm_factor` in all of the attention backends.

What do you think @cyanguwa?

Reply:
I think keeping things as `softmax_scale = 1/sqrt(head_dim)` is good!